### ISSUE TITLE
Creation of furniture and parametric components for the FreeCAD library - Time Commitment Grant Proposal

### ISSUE THREAD

**Issue by Francisco-Rosa** [FreeCAD/FPA-grant-proposals#5]
### Proposal description

CAD/BIM programs generally have a good library of components ready for application in the development of architectural and engineering projects.

FreeCAD already has a library, but it is necessary to increase the supply of components, both in their number and in their further development, mainly in a greater degree of parameterization.

The availability of this resource is always a significant requirement when choosing the application to use in project development, both by professionals and companies. This may increase interest in using FreeCAD.

Since the FreeCAD component library is public, everyone using this platform would benefit from access to these resources.

### Deliverables

The idea is to develop and supply a few dozen pieces of furniture and parametric components (for architectural and engineering projects) for immediate use in FreeCAD.

This will enable better development and detailing of projects in these areas, aiming to improve compliance with BIM criteria.

### Timeline

I can start as soon as my grant proposal is accepted.

I can spend 4 hours days a week for the next 5 months working on this project.

### Risks and mitigation

As a mitigating measure, I present the possibility of compensating for any missing hours.

### Compensation

This work would have a value of U$5,000 (R$25,000.00).
The value of the hour worked would be around U$45.00 (R$225.00), paid in five monthly installments of U$1,000.00.

### About you

Name:
Francisco de Assis Rosa

Forum / Community username(s):
F_Rosa

Experience relevant to the proposed grant:
I have already collaborated with the creation of a component library: https://github.com/Francisco-Rosa/FreeCAD-library.

Have you already contributed to FreeCAD? Please elaborate:
In addition to the library mentioned, I developed and maintains Movie Workbench: https://github.com/Francisco-Rosa/FreeCAD-Movie; and several posts on the FC Forum, highlighting:
Topography with point cloud:
https://forum.freecad.org/viewtopic.php?f=8&t=58925
Urbanization:
https://forum.freecad.org/viewtopic.php?t=59084
Rendering:
https://forum.freecad.org/viewtopic.php?t=82656
BIM:
https://forum.freecad.org/viewtopic.php?t=60050
And
https://forum.freecad.org/viewtopic.php?t=59192
Animations:
https://forum.freecad.org/viewtopic.php?t=74432
2D Drawing:
https://forum.freecad.org/viewtopic.php?p=581760#p581760
Creation of doors and windows:
https://forum.freecad.org/viewtopic.php?f=24&t=18475&p=144393#p144393
Foundations, AC installations, compatibility and libraries:
https://forum.freecad.org/viewtopic.php?t=19025
Parametric glass skin:
https://forum.freecad.org/viewtopic.php?p=140609#p140609

Have you ever worked on similar software before? Please elaborate:
In addition to the FreeCAD, that I currently use, I have worked with several commercial and open source programs, such as:
     • BIM (ArchiCAD and Revit);
     • 3D CAD (ArchiCAD, Artlantis, Blender, Form Z and Microstation);
     • 2D CAD (AutoCAD, DaftSight and LibreCAD).

Have you ever solved similar tasks before? Please elaborate:
As already mentioned, I created several components for FreeCAD: https://github.com/Francisco-Rosa/FreeCAD-library.

**Comment by chennes** [comment]
Thank you for the proposal, @Francisco-Rosa. The FPA Grant Review Committee has begun evaluating grant proposals, and suggests that you add an additional statement to your proposal regarding documentation of the objects and their parameterization.

**Comment by yorikvanhavre** [comment]
I would also like if you could do a quick, rough list of the models you think you could make. Even if that list changes afterwards, no problem, but so we have a kind of metrics to start with

**Comment by Francisco-Rosa** [comment]
**Objects and their parameterization**

The work consists of the elaboration and/or review of components listed below, built by solid and native FreeCAD objects, parameterized (modulations, dimensions, and materials) and, when applicable, accompanied by 2D representations (plan and elevations), rendered image and brief descriptive and explanatory editable memorial of the component.

For this, 3 phases of activities are planned:

> **Phase 1** - Review of some existing components listed below that can be reused, expected duration: 1 month.
> 
> **Phase 2** - Creation of new components, expected duration: 3 months.
> 
> **Phase 3** - Preparation of 2D representations, rendered images and memorials, expected duration: 1 month.

**Component list**

The total number of components prepared may exceed the list mentioned below. However, to exemplify and for a minimum prevision, we describe below the numbers of components with the predictions indicated in parentheses, totaling a quantity between 57 and 90 components:

1. Review of building components (20 to 34 components) - phases 1 and 3.

> 1.1. Doors, windows, and frames: general review to meet the objectives described previously. (10 to 20 components).
> 1.2. Foundations and HVAC: general review to meet the objectives described previously. (10 to 14 components).

2. Residential and office furniture (37 to 56 components), phases 2 and 3.

> 2.1 Living and dining room (9 to 17 components): sofas (1 to 2), armchairs (1 to 2), chairs (1 to 2), tables (1 to 2), rug (1), painting (1) , lamps (1 to 2), vases (1 to 2), shelves (1 to 2) and curtain (1).
> 2.2. Kitchens (8 to 15 components): cabinet (1), worktop (1), sink (1), taps (1 to 2), stove (1), fridge (1), freezer (1) and hood (1).
> 2.3. Bathrooms (7 to 8 components): toilet (1), shower (1), basin (1), taps (1 to 2), bathtub (1), closet (1) and shower box (1).
> 2.4. Service areas (4 components): laundry sink (1), washing machines (1) and dryer (1) and tap (1).
> 2.5. Work/office areas (4 to 5 components): tables (1 to 2), chair (1), computer equipment (1) and table lamp (1).
> 2.6. Gardens (2 to 4 components): plants (1 to 2) and pots (1 to 2).
> 2.7. Garages (3 components): vehicle (1), bicycle (1) and gate (1).

We remain at your disposal for further clarification.

**Comment by chennes** [comment]
@Francisco-Rosa the FPA is having some discussion about this proposal during the voting process, and there will be a short delay before we have made a final decision -- thank you for your patience.

**Comment by chennes** [comment]
@Francisco-Rosa it doesn't look like you've committed anything to https://github.com/Francisco-Rosa/FreeCAD-library in a long time -- can you post a few links to recently-created FCStd models?

**Comment by yorikvanhavre** [comment]
Would you share one of these FCStd files to have a look? (Those in that repo are jpg files only).
Just to help the reviewing team to assess what we'd get...

**Comment by Francisco-Rosa** [comment]
As requested, I inserted an example of what is intended to be accomplished. The component configurations are made through the properties window of its insertion point (there is a video showing this), this is done through links, clones or simple copies.

Greetings.

https://github.com/Francisco-Rosa/Recently-created-FCStd-models/tree/main/Example

**Comment by yorikvanhavre** [comment]
That's perfect! Thanks a lot!

**Comment by shaise** [comment]
Indeed it is very nice! Maybe the reliance on Render workbench should be removed because its not by default?

**Comment by Francisco-Rosa** [comment]
Yes, I let the projects render as an example. But it would be interesting if we left some materials pre-defined, if possible.

**Comment by yorikvanhavre** [comment]
The general idea of the new Materials system is that it supports everything, render included. So soon we can start giving materials that are render-ready

**Comment by chennes** [comment]
@Francisco-Rosa the FPA members have voted to approve funding for this project. Please send an email to fpa@freecad.org confirming that you are ready to begin. We will then coordinate reporting and payment details with you.

**Comment by yorikvanhavre** [comment]
This grant has now been completed. @Francisco-Rosa , is everything already part of the Parts Library, or are there still pieces missing?

**Comment by yorikvanhavre** [comment]
Sorry for that, I suppose it went under the radar. It's merged now!
Thanks for the work there @Francisco-Rosa !

This grant is now completed, so I'm closing the issue.
